### DISCUSSION TITLE
SOAPHeaderElementProcessorWSS4J: Ability to add custom WSSConfig processors

### DISCUSSION THREAD

**Discussion by sopgreg** [phax/phase4#146] [General]
Is there currently a way to add a custom processor to the `WSSConfig` in `SOAPHeaderElementProcessorWSS4J#_verifyAndDecrypt`?

This is necessary to add a custom processor for the `EncryptedKey` part, since WSS4J does not support parts for EC, like so:

`aWSSConfig.setProcessor(new QName("http://www.w3.org/2001/04/xmlenc#", "EncryptedKey"), customProcessor);`

**Comment by phax** [comment]
Not at the moment - but I understand the request. WIll be added.

**Reply by phax** [reply]
Created issue #150 from it

**Reply by phax** [reply]
Implemented as part of the 2.2.0 release using `IAS4DecryptParameterModifier` interface - overload the `modifyWSSConfig` to achieve your goal
